src/autopas/LogicHandler.h

```diff
@@ -520,37 +526,7 @@ class LogicHandler {
    * @param haloParticleBuffers
    */
   void setParticleBuffers(const std::vector<FullParticleCell<Particle>> &particleBuffers,
-                          const std::vector<FullParticleCell<Particle>> &haloParticleBuffers) {
-    auto exchangeBuffer = [](const auto &newBuffers, auto &oldBuffers, auto &particleCounter) {
-      // sanity check
-      if (oldBuffers.size() < newBuffers.size()) {
-        autopas::utils::ExceptionHandler::exception(
-            "The number of new buffers ({}) is larger than number of existing buffers ({})!", newBuffers.size(),
-            oldBuffers.size());
-      }
-
-      // we will clear the old buffers so subtract the particles from the counters.
-      const auto numParticlesInOldBuffers =
-          std::transform_reduce(oldBuffers.begin(), std::next(oldBuffers.begin(), newBuffers.size()), 0, std::plus<>(),
-                                [](const auto &cell) { return cell.numParticles(); });
-      particleCounter.fetch_sub(numParticlesInOldBuffers, std::memory_order_relaxed);
-
-      // clear the old buffers and copy the content of the new buffers over.
-      size_t numParticlesInNewBuffers = 0;
-      for (size_t i = 0; i < newBuffers.size(); ++i) {
-        oldBuffers[i].clear();
-        for (const auto &p : newBuffers[i]) {
-          ++numParticlesInNewBuffers;
-          oldBuffers[i].addParticle(p);
-        }
-      }
-      // update the counters.
-      particleCounter.fetch_add(numParticlesInNewBuffers, std::memory_order_relaxed);
-    };
-
-    exchangeBuffer(particleBuffers, _particleBuffer, _numParticlesOwned);
-    exchangeBuffer(haloParticleBuffers, _haloParticleBuffer, _numParticlesHalo);
-  }
+                          const std::vector<FullParticleCell<Particle>> &haloParticleBuffers);
```

Comment:
If there are `.cpp` and `.h` and the class / function is not templated -> always declare it in the header and implement it in the `.cpp`.

If it is templated I think it is better to split declaration and implementation within the header. Otherwise, you have to make sure that you do not use any functions that are only declared further down. Also it produces a cleaner look if you separate declarations + doc vs implementation. One can make the argument that this doesn't really matter for trivial functions such as getter/setter, hence I wouldn't be too harsh about this. In general I try to split declarations and implementations (but sometimes I accidentally don't so feel free to call me out :D )